hbase-server/src/test/java/org/apache/hadoop/hbase/replication/regionserver/TestWALEntryStream.java

```diff
@@ -103,6 +107,27 @@
   private static final NavigableMap<byte[], Integer> scopes = getScopes();
   private final String fakeWalGroupId = "fake-wal-group-id";
 
+  /**
+   * Test helper that waits until a non-null entry is available in the stream next or times out.
+   */
+  private static class WALEntryStreamWithRetries extends WALEntryStream {
```

Comment:
Can u also pls explain why we use this special subclass here for the tests?  (as comments) will be easy to read and digest later.

Reply:
Addressed in #3344